src/Space.ts

```diff
@@ -92,11 +92,11 @@ class Space extends EventTarget {
     const presence = this.channel.presence;
 
     // TODO: Discuss if we actually want change this behaviour in contrast to presence (enter becomes an update)
-    presence.get({ clientId }).then(
+    return presence.get({ clientId }).then(
       (presenceMessages) =>
         new Promise((resolve, reject) => {
           if (presenceMessages && presenceMessages.length === 1) {
-            reject(ERROR_CLIENT_ALREADY_ENTERED);
+            reject(new Error(ERROR_CLIENT_ALREADY_ENTERED));
```

Comment:
What's the policy for error responses from this API? Are we going to mirror what we have in our existing APIs, with specific error codes etc?

Reply:
That's the plan @paddybyers, yes. However, the above was prematurely implemented, after a discussion we decided to remove this behaviour as it wasn't adding any value (i.e. we keep the idempotency of these presence methods. For UI frameworks, if this is found to be the root of unnecessary re-renders, we can handle it in the framework-specific code)